src/browser.ts

```diff
@@ -138,11 +140,37 @@ export class GrpcClient {
     configOverrides: gax.ClientConfig,
     headers: OutgoingHttpHeaders
   ) {
+    function buildMetadata(abTests, moreHeaders) {
+      const metadata = {};
+      if (moreHeaders) {
+        for (const key in moreHeaders) {
+          if (
+            key.toLowerCase() !== 'x-goog-api-client' &&
+            moreHeaders.hasOwnProperty(key)
+          ) {
+            const value = moreHeaders[key];
+            if (Array.isArray(value)) {
+              if (metadata[key] === undefined) {
+                metadata[key] = value;
+              } else {
+                value.forEach(v => {
+                  metadata[key].push(v);
+                });
+              }
+            } else {
+              metadata[key] = value;
```

Comment:
If `value` is not an array, it makes sense to do `metadata[key] = [value];` so that they are all arrays in the metadata.